portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -77,17 +78,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String username = request.getParameter("username");
     String commentText = request.getParameter("text");
+    String sanitizedUsername = HtmlEscapers.htmlEscaper().escape(username);
```

Comment:
What is the purpose of the Escaper here?

Reply:
The rationale is that an attacker could also conduct HTML injection through the username field, which is also rendered client-side. This protects from dangerous usernames getting rendered to the page

Reply:
That's actually very thoughtful. However, html injection is more of a client side attack. It might be an overkill for server codes to do escaping for everything that might be displayed on the html.

To achieve the same effect and save you these effect, try to set textContext property instead of innerHTML: https://www.w3schools.com/jsref/prop_node_textcontent.asp
The browser treats this property as string literal while rendering

Reply:
ok, that makes sense. Should I keep the HTML escaping on server side in this case?

Reply:
No, please remove the escaping. It is usually not a good practice to silently change user's input.

Reply:
Done